dot/parachain/backing/second.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package backing
+
+import (
+	"errors"
+	"fmt"
+
+	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+var (
+	errWrongPVDForSecondingCandidate = errors.New(
+		"incorrect persisted validation data provided for seconding candidate")
+	errUnknownRelayParentForSecondingCandidate = errors.New(
+		"attempted to second a candidate outside of our view")
+	errParaOutsideAssignmentForSeconding = errors.New(
+		"subsystem requested to second for parachain beyond our assignment scope")
+	errAlreadySignedValidStatement = errors.New("already signed a valid statement for this candidate")
+)
+
+func (cb *CandidateBacking) handleSecondMessage(
+	candidateReceipt parachaintypes.CandidateReceipt,
+	pvd parachaintypes.PersistedValidationData,
+	pov parachaintypes.PoV,
+	chRelayParentAndCommand chan relayParentAndCommand,
+) error {
+	pvdBytes, err := scale.Marshal(pvd)
+	if err != nil {
+		return fmt.Errorf("marshalling persisted validation data: %w", err)
+	}
+
+	pvdHash, err := common.Blake2bHash(pvdBytes)
+	if err != nil {
+		return fmt.Errorf("hashing persisted validation data: %w", err)
+	}
+
+	if candidateReceipt.Descriptor.PersistedValidationDataHash != pvdHash {
+		return errWrongPVDForSecondingCandidate
```

Comment:
I would say that given the error context can be useful for log informations, for example in this case if `A != B` then we just return `errWrongValue` but will be better if we return `errWrongValue: expected A got B`

```suggestion
		return fmt.Errorf("%w: mismatch between %v and %v", errWrongPVDForSecondingCandidate, 
		  candidateReceipt.Descriptor.PersistedValidationDataHash,
		  pvdHash)
```